### PR TITLE
[0.3.0] Club Joining & Invitation Endpoints

### PR DESCRIPTION
**Pull request by Galax028**
I am too lazy to write this, please review!

### CODE REVIEW

**Comment by Galax028** [comment]
I can't believe I worked on this release using only GitHub Codespaces :skull:

**Comment by Galax028** [comment]
@owo93 I know I didn’t request your review but, please read my changes on your assigned files so you get a better idea of how this works!

File: mysk-data-api/src/routes/v1/clubs/add_club_members.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        _ => unreachable!("Club::get_by_id should always return a Detailed variant"),
+    };
+
+    // Check if the inviting student is a staff of the club
+    if !club.staffs.iter().any(|staff| match staff {
+        Student::IdOnly(staff, _) => staff.id == inviter_student_id,
+        _ => unreachable!("Staff should always be an IdOnly variant"),
+    }) {
+        return Err(Error::InvalidPermission(
+            "Student must be a staff of the club to create contacts".to_string(),
```
Comment (smartwhatt) [inline mysk-data-api/src/routes/v1/clubs/add_club_members.rs:104]: Wrong error detail

**Review by smartwhatt — APPROVED** [review]
LGTM but fix the error message before merging

**Pull request merged**